### PR TITLE
Docker build fix

### PR DESCRIPTION
**Pull request by dingsheng-ong**
I'm trying to fix the docker build errors. The built image can also be found on Docker Hub.
`docker pull sheng970303/aimlackies-reporter:latest`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -3,9 +3,37 @@ FROM ubuntu:20.04
...
+    "CREATE USER IF NOT EXISTS 'user0'@'%' IDENTIFIED BY 'secret';\n \
+    GRANT ALL PRIVILEGES ON * . * TO 'user0'@'%';\n \
+    FLUSH PRIVILEGES;" \
+ && mysql -u "root" -e "CREATE DATABASE IF NOT EXISTS local_reporter;"
+# write environment variables
+RUN echo -n \
+"export AIMLAC_CC_MACHINE=34.72.51.59\n\
+export AIMLAC_API_KEY=AIMLACkies275001901\n\
+export AIMLACKIES_REPORTER_SECRET_KEY=W5Jal+qx9vOh8gjaugNZPg==\n\
+export AIMLACKIES_REPORTER_SECURITY_PASSWORD_SALT=288872824872550364092124909785067926279\n\
```
Comment (fjebaker) [inline Dockerfile]: Should these be environment variables passed that are passed in at runtime? Along the lines of
```bash
docker run --env AIMLAC_CC_MACHINE=$AIMLAC_CC_MACHINE ...
```
or via a `.env` file?

**Review by fjebaker — APPROVED** [review]
LGTM

File: Dockerfile (hunk 1 above)
Comment (dingsheng-ong) [reply Dockerfile:25]: Aimlackies didn't look at environment variables to override those settings while running the server. So, passing `—env` or `—env-file` to `docker run` basically has no effect. What I can do is make those variables build args for `docker build`, so we can modify them with the line `docker build —build-arg AIMLAC API KEY=XXX...`. But we'd have to rebuild the image every time we changed the variables. Another ugly hack is to bind the new config file and replace the old bash profile with the new file: `docker run -v <new-file>:/root/.bash_profile ...`.

File: Dockerfile [hunk 2]
```diff
@@ -18,12 +18,16 @@ RUN service mysql start \
...
 RUN echo -n \
 "export AIMLAC_CC_MACHINE=34.72.51.59\n\
-export AIMLAC_API_KEY=AIMLACkies275001901\n\
-export AIMLACKIES_REPORTER_SECRET_KEY=W5Jal+qx9vOh8gjaugNZPg==\n\
-export AIMLACKIES_REPORTER_SECURITY_PASSWORD_SALT=288872824872550364092124909785067926279\n\
-export AIMLACKIES_REPORTER_DATABASE_URL=mysql+pymysql://user0:secret@localhost/local_reporter\n\
+export AIMLAC_API_KEY=${AIMLAC_API_KEY}\n\
+export AIMLACKIES_REPORTER_SECRET_KEY=${AIMLACKIES_REPORTER_SECRET_KEY}\n\
+export AIMLACKIES_REPORTER_SECURITY_PASSWORD_SALT=${AIMLACKIES_REPORTER_SECURITY_PASSWORD_SALT}\n\
+export AIMLACKIES_REPORTER_DATABASE_URL=${AIMLACKIES_REPORTER_DATABASE_URL}\n\
```
Comment (dingsheng-ong) [inline Dockerfile:30]: I've made these 4 variables as build arguments, please include `--build-arg AIMLAC_API_KEY=XXX` when building the image

File: Dockerfile (hunk 1 above)
Comment (dingsheng-ong) [reply Dockerfile]: the latest commit now allows us to pass the variables at runtime. Either way will do:
```
docker run --env AIMLAC_API_KEY=AIMLACkies275001901 ...
```
OR
```
docker run --env-file envfile ...
```
:warning: **NOTE**: the latest commit does not include the `AIMLAC_API_KEY`, so you need to include the API keys when you run the image.

**Review by fjebaker — APPROVED** [review]
LGTM!

**Pull request merged**